mobile_config/src/gateway_info_v3.rs

```diff
@@ -0,0 +1,282 @@
+
+                    map.insert(
+                        PublicKeyBinary::from_str(&entity_key)?,
+                        MobileTrackerInfo {
+                            location: asserted_location.map(|v| v as u64),
+                            last_changed_at,
+                            asserted_location_changed_at,
+                        },
+                    );
+                    Ok(map)
+                },
+            )
+            .await
+    }
+
+    // Streams all gateway info records, optionally filtering by device types.
+    pub fn all_info_stream_v3<'a>(
+        db: impl PgExecutor<'a> + 'a,
+        device_types: &'a [DeviceTypeV2],
+        mtim: &'a MobileTrackerInfoMap,
+    ) -> impl Stream<Item = GatewayInfoV3> + 'a {
+        // Choose base query depending on whether filtering is needed.
+        let query = if device_types.is_empty() {
+            sqlx::query(GET_MOBILE_HOTSPOT_INFO)
+        } else {
+            sqlx::query(&DEVICE_TYPES_METADATA_SQL).bind(
+                device_types
+                    .iter()
+                    // The device_types field has a jsonb type but is being used as a string,
+                    // which forces us to add quotes.
+                    .map(|v| format!("\"{}\"", v.to_sql_param()))
+                    .collect::<Vec<_>>(),
+            )
+        };
+
+        query
+            .fetch(db)
+            .filter_map(move |result| async move {
+                match result {
+                    Ok(row) => process_row(row, mtim).await,
+                    Err(e) => {
+                        tracing::error!("SQLx fetch error: {e:?}");
+                        None
+                    }
+                }
+            })
+            .boxed()
+    }
+
+    // Processes a single database row into a GatewayInfoV3, returning None if any step fails.
+    async fn process_row(
+        row: sqlx::postgres::PgRow,
+        mtim: &MobileTrackerInfoMap,
+    ) -> Option<GatewayInfoV3> {
+        let device_type = DeviceTypeV2::from_str(
+            row.get::<Json<String>, &str>("device_type")
+                .to_string()
+                .as_ref(),
+        )
+        .map_err(|err| sqlx::Error::Decode(Box::new(err)))
+        .ok()?;
+
+        let address = PublicKeyBinary::from_str(
+            &bs58::encode(row.get::<&[u8], &str>("entity_key")).into_string(),
+        )
+        .map_err(|err| sqlx::Error::Decode(Box::new(err)))
+        .ok()?;
+
+        let mti = mtim.get(&address)?;
+
+        let updated_at = mti.last_changed_at;
+
+        let metadata = mti.location.and_then(|loc| {
+            let location_changed_at = mti.asserted_location_changed_at?;
+            let deployment_info = row
+                .try_get::<Option<sqlx::types::Json<DeploymentInfo>>, _>("deployment_info")
+                .ok()
+                .flatten()
+                .and_then(|json| match json.0 {
```

Comment:
`sqlx::types::Json` provides `Deref` to it's inner type. We don't need to access the inner member here with `json.0`.